### PR TITLE
API-385 Update queries to use new regions tables

### PR DESCRIPTION
**Pull request by rrequero**
Update queries of current events in production to use new regions table world-fishing-827.pipe_regions_layers.event_regions_v20220706.

### CODE REVIEW

File: assets/bigquery/encounter-events.sql.j2 [hunk 1]
```diff
@@ -105,7 +123,46 @@ WITH
...
+  ),
+  event_by_layer_by_id as (
+    select 
+      event_id, layer, id
+    from create_event_region_matches
+    group by 1, 2, 3
+  ),
+  events_by_layer as(
+    select
+      event_id, CONCAT('"', layer, '":' ,TO_JSON_STRING(array_agg(id))) as json_frament
```
Comment (pwoods25443) [inline assets/bigquery/encounter-events.sql.j2:143]: typo in 'json_frament' should be 'json_fragment'.   That typo comes from my original code :-)

**Review by pwoods25443 — APPROVED** [review]
Looks ok to me.  Something to consider for the future is to put the region matching code into a separate j2 file and then {% include %} it in each SQL file so that it will be easier to change in the future.

File: assets/bigquery/encounter-events.sql.j2 (hunk 1 above)
Comment (rrequero) [reply assets/bigquery/encounter-events.sql.j2]: Fixed

**Review by aperdizs — APPROVED** [review]
LGTM 

I like the improvement that Paul suggested. I think it's time to change it because, otherwise, maybe we'll never do it,

Another minor thing and just from my point of view, I would change the `fe_regions` query name for another a little more descriptive. F.i. `fishing_events_with_regions`.

**Comment by rrequero** [comment]
@pwoods25443 @aperdizs I did the refactor to generate a template to include the subquery of regions table.

**Pull request merged**